File: src/test/java/s3/TestS3.java
package s3;

import oopsla25.TestUtil;
import org.junit.Test;

public class TestS3 {

    @Test
    public void testWFAmbiguous() {
        TestUtil.testBad(TestUtil.BASE_PATH + "\\s3\\bad\\WFAmbiguous.scr");
    }

    @Test
    public void testLabelCapture() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s3\\good\\LabelCapture.scr");
    }

    @Test
    public void testRoleCapture() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s3\\good\\RoleCapture.scr");
    }

    @Test
    public void testProgress() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s3\\good\\Progress.scr");
    }

    @Test
    public void testClearTermination1() {
        TestUtil.testBad(TestUtil.BASE_PATH + "\\s3\\bad\\ClearTermination1.scr");
    }

    @Test
    public void testClearTermination2() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s3\\good\\ClearTermination2.scr");
    }

    @Test
    public void testClearTermination3() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s3\\good\\ClearTermination3.scr");
    }

    @Test
    public void testBalance() {
        TestUtil.testBad(TestUtil.BASE_PATH + "\\s3\\bad\\Balance.scr");
    }
}

File: src/test/scala/com/github/rhu1/gt/test/oopsla25/s3/TestS3.scala
package com.github.rhu1.gt.test.oopsla25.s3;

import com.github.rhu1.gt.test.oopsla25.TestUtil
import org.scalatest.funsuite.AnyFunSuite;


class TestS3 extends AnyFunSuite {

    test("WFAmbiguous") {
        TestUtil.testBad(TestUtil.BASE_PATH + "/s3/bad/WFAmbiguous.scr");
    }

    test("LabelCapture") {
        TestUtil.testGood(TestUtil.BASE_PATH + "/s3/good/LabelCapture.scr");
    }

    test("RoleCapture") {
        TestUtil.testGood(TestUtil.BASE_PATH + "/s3/good/RoleCapture.scr");
    }

    /*test("Progress") {
        TestUtil.testGood(TestUtil.BASE_PATH + "/s3/good/Progress.scr");
    }*/

    test("ClearTermination1") {
        TestUtil.testBad(TestUtil.BASE_PATH + "/s3/bad/ClearTermination1.scr");
    }

    test("ClearTermination2") {
        TestUtil.testGood(TestUtil.BASE_PATH + "/s3/good/ClearTermination2.scr");
    }

    test("ClearTermination3") {
        TestUtil.testGood(TestUtil.BASE_PATH + "/s3/good/ClearTermination3.scr");
    }

    test("Balance") {
        TestUtil.testBad(TestUtil.BASE_PATH + "/s3/bad/Balance.scr");
    }
}
